src/interfaces/chain.cpp

```diff
@@ -263,6 +264,19 @@ class ChainImpl : public Chain
         return std::move(result);
     }
     std::unique_ptr<Chain::Lock> assumeLocked() override { return MakeUnique<LockImpl>(); }
+    bool filterMatchesAny(const uint256& hash, const GCSFilter::ElementSet& filter_set) override
```

Comment:
Maybe make the BlockFilterType a parameter here?

Reply:
Will there ever be a filter different from the basic filter?